code/scripts/DataTableGen.hs

```diff
@@ -0,0 +1,292 @@
+  nlEntry = makeEntry "" "newline" "" [] [] [] []
+-- creates actual file entries
+createEntry homeDirectory file filename = do
+  let drpk = DC.fileDrasilPack file
+      fn = filename
+      fp = DC.filePath file
+      -- entry file path is stripped of home directory and drasil- pack
+      efp = (++ "/") $ (\\ homeDirectory ++ "/drasil-" ++ drpk ++ "/") fp
+
+  -- extracts entry data from File data type
+  -- stripInstances = [(dataType,classInfo)]
+  (dtl,ntl,classNames,stripInstances) <- SCR.extractEntryData fn fp
+  
+  let clstp
+        | drpk == "gool" = GOOL
+        | otherwise = Drasil
+
+  let cls = map (makeClass clstp) classNames
+      clsint = map makeClassInstance stripInstances
+
+  let entry = makeEntry drpk fn efp dtl ntl cls clsint
+  return entry
+
+-- creates entrystring for each entry (contains lines for each entry's data)
+compileEntryData :: [ClassName] -> Entry -> DC.FileName -> IO EntryString
+-- creates blank line entrystring to separate file entries by drasil- package
+compileEntryData ordClassInsts entry "newline" = return "\t"
+-- creates entrystrings for actual file entries
+compileEntryData ordClassInsts entry filename = do
+
+  -- joins data, newtype and class values/placeholders (first data entry line)
+  let f d n c = intercalate "," [drpk,fp,fn,d,n,c]
+      (drpk,fp,fn) = (drasilPack entry,filePath entry,fileName entry)
+
+  -- extracts data, newtype and class names + class instances from entry
+  let dataNames = dataTypes entry
+      newtypeNames = newtypes entry
+      classNames = map className (classes entry)
+      entryClassInsts = classInstances entry
+
+  -- guards determine how to handle first data entry line
+  let entry 
+        | not (null dataNames) = f (head dataNames) "\t" "\t"
+        | not (null newtypeNames) = f "\t" (head newtypeNames) "\t"
+        | not (null classNames) = f "\t" "\t" (head classNames)
+        | otherwise = f "\t" "\t" "\t"
+  
+  -- creates heads of each data, newtype and class entry line
+  let dtEntryHds = map (("\t,\t,\t,"++) . (++",\t,\t")) dataNames
+      ntEntryHds = map (("\t,\t,\t,\t,"++) . (++",\t")) newtypeNames
+      clsEntries = map ("\t,\t,\t,\t,\t,"++) classNames
+
+  -- creating "Y" references to class instances for data types
+  let dtInstRefNames = map (getRefNames entryClassInsts) dataNames
+      dtRefLines = createRefLines ordClassInsts dtInstRefNames
+      dtEntries = zipWith join' dtEntryHds dtRefLines
+
+  -- creating "Y" references to class instances for newtype types
+  let ntInstRefNames = map (getRefNames entryClassInsts) newtypeNames
+      ntRefLines = createRefLines ordClassInsts ntInstRefNames
+      ntEntries = zipWith join' ntEntryHds ntRefLines
+
+  -- creates file entry data by combining data, newtype and class entries
+  let entryData = dtEntries ++ ntEntries ++ clsEntries
+  
+  -- guards determine how to handle overall file entry output
+  -- for single line entry data vs. multi-line entry data
+  let output 
```

Comment:
Here too, you should use pattern-matching.

Reply:
I refactored the definition for `output` [here](https://github.com/JacquesCarette/Drasil/pull/2208/commits/4065330bf5885d36437fb97d322a5c23d339df61#) (old lines 205-214; new lines 204-213).